### ISSUE TITLE
compare modifyed workspace with Head SVN revison

### ISSUE THREAD

**Issue by Vintic** [jenkinsci/last-changes-plugin#60]
Hi.
After build, I need to check what changes build made comparing with last revision.

Why I need it.
Before I deploy release to production, I need to check if was made some hot fixes directly on production without committing them in SVN.
What I think to do.
I will download that changes using RSync to Jenkins workspace and check modification with Last Changes Plugin. If all is OK I will deploy release to production.

**Comment by rmpestano** [comment]
Hi, the plugin needs the source code to be commited to a VCS because It uses source control commands to create the diff like svn diff or git diff.

About your use case I think you just need a delivery pipeline which has a stage where you can check the changes before going to next stage that can be the deploy to production.

**Comment by Vintic** [comment]
ok, but it is possible to compare files from work space with those from head revision like SVN Diff make?
For example if, was specified head revision in Specific revision section, to use this code: 
`diff.setSources(SvnTarget.fromURL(new File(lastCommitInfo.getCommitFilePath())),
                    SvnTarget.fromURL(repository.getLocation(), SVNRevision.create(currentRevision)));`

**Comment by rmpestano** [comment]
As far as I know (I may be wrong) the diff is done between two revisions and you don't have a revision until you commit to the repository.

Anyway even if possible I don't plan to add support for such usecase unless the fix is transparent to enduser (e.g no UI change).

**Comment by Vintic** [comment]
Being in checkout directory and writing in command line " svn diff --revision 26347 "
it display local changes that was made.

**Comment by Vintic** [comment]
Here is implementation with SVNDiffClient:
https://wiki.svnkit.com/Diffing%20a%20working%20copy%20against%20a%20repository

**Comment by rmpestano** [comment]
I see, I still think it is a very "strange" use case to support. 

I'd suggest you tu fork this plugin and add this feature you need. You can see build and run [instructions here](https://github.com/jenkinsci/last-changes-plugin#10-running-it-locally).

**Comment by Vintic** [comment]
Thank you, 

I found a solution and made some changes in code: 
Now when I write in "Specific revision" word: "Base".

Is executing something like:
`lastChanges = svnLastChanges.changesOf(repository, SVNRevision.WORKING, SVNRevision.BASE);`

and some minor changes was added to work properly.

**Comment by Vintic** [comment]
Hi, again)

How I can add file description ( like: modified date ) in the header of file diff div.

Usage:
I want to highlight files that was modified on production from last build, by checking file modified date and last success build date.

**Comment by rmpestano** [comment]
Hi,

the file list of changed files is generated by [diff2html](https://github.com/rtfpessoa/diff2html), you may ask there if there's a way to add info to the list. 

You probably know but the UI code resides in THIS jelly file: https://github.com/jenkinsci/last-changes-plugin/blob/50814d8d722c89a159b440ab60cdd145696745c7/src/main/resources/com/github/jenkins/lastchanges/LastChangesBuildAction/index.jelly

**Comment by SunBlack** [comment]
I agree with @Vintic that it would be nice to be able to compare the working directory with the last commit. This way it would be possible to show changes by clang-format or other code formatting tools for example (and in best case the tool can also set the build failure state if there is a change).
